todo-app/src/Task.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import AppTask from './AppTask';
+import AppDate from './AppDate';
+import './index.css';
+
+export default class Task extends React.Component{
+
+  handleTaskDone = () =>  this.props.handleTaskDone(this.props.todoItem.id)
+  
+  render() {
+  const{todoItem, handleTaskDone} = this.props;
+       return (
+            <li className="task-list">
+             <span>{todoItem.done ? <input type = 'checkbox' onClick={()=>handleTaskDone(todoItem.id)}/>
+              :
+              <input type = 'checkbox' checked onClick={()=> handleTaskDone(todoItem.id)}/>   
+             }
+                 </span>
+            <span> {todoItem.done? <span className = 'task-done'>
```

Comment:
You've got a few problems here.

First of all, your `todoItem.done` value is displaying backwards - you are showing an empty checkbox when `todoItem.done` is `true`, and a full checkbox when it's `false`.

Second, if you run your app and open the Chrome console and click a checkbox, you'll see:

    proxyConsole.js:56 Warning: Task is changing an uncontrolled input of type checkbox to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components

You can check out the link provided in that warning to see what this is all about. Basically, you are first providing an uncontrolled input (a checkbox with no `checked` attribute specified), and then replacing it with a controlled input (a checkbox with a `checked` attribute set to true, ie it can never be unchecked).

So, what's the root of the problem? You are using a ternary to split rendering based on the `todoItem.done` value, which controls whether or not a checkbox is checked. But this is unnecessary. The "React" way to do this is:

    <input type='checkbox' checked={todoItem.done} onClick={() => handleTaskDone(todoItem.id)} />

No ternary, just pass the checked value directly.